cudaextender/src/ungapped_xdrop_kernels.cu

```diff
@@ -0,0 +1,493 @@
+        //Left extension
+
+        if (lane_id == 0)
+        {
+            tile[warp_id]           = 0;
+            xdrop_found[warp_id]    = false;
+            edge_found[warp_id]     = false;
+            new_max_found[warp_id]  = false;
+            prev_score[warp_id]     = 0;
+            prev_max_score[warp_id] = -1000;
+            prev_max_pos[warp_id]   = 0;
+            left_extent[warp_id]    = 0;
+        }
+
+        count_del[0] = 0;
+        count_del[1] = 0;
+        count_del[2] = 0;
+        count_del[3] = 0;
+        max_pos      = 0;
+        __syncwarp();
+
+        while (!xdrop_found[warp_id] && !edge_found[warp_id])
+        {
+            pos_offset   = lane_id + 1 + tile[warp_id];
+            thread_score = 0;
+
+            if (ref_loc[warp_id] >= pos_offset && query_loc[warp_id] >= pos_offset)
+            {
+                ref_pos      = ref_loc[warp_id] - pos_offset;
+                query_pos    = query_loc[warp_id] - pos_offset;
+                r_chr        = d_target[ref_pos];
+                q_chr        = d_query[query_pos];
+                thread_score = sub_mat[r_chr * nuc + q_chr];
+            }
+
+#pragma unroll
+            for (int32_t offset = 1; offset < warp_size; offset = offset << 1)
+            {
+                temp = __shfl_up_sync(0xFFFFFFFF, thread_score, offset);
+
+                if (lane_id >= offset)
+                {
+                    thread_score += temp;
+                }
+            }
+
+            thread_score += prev_score[warp_id];
+            if (thread_score > prev_max_score[warp_id])
+            {
+                max_thread_score = thread_score;
+                max_pos          = pos_offset;
+            }
+            else
+            {
+                max_thread_score = prev_max_score[warp_id];
+                max_pos          = prev_max_pos[warp_id];
+            }
+            __syncwarp();
+
+#pragma unroll
+            for (int32_t offset = 1; offset < warp_size; offset = offset << 1)
+            {
+                temp     = __shfl_up_sync(0xFFFFFFFF, max_thread_score, offset);
+                temp_pos = __shfl_up_sync(0xFFFFFFFF, max_pos, offset);
+
+                if (lane_id >= offset)
+                {
+                    if (temp >= max_thread_score)
+                    {
+                        max_thread_score = temp;
+                        max_pos          = temp_pos;
+                    }
+                }
+            }
+
+            xdrop_done = ((max_thread_score - thread_score) > xdrop_threshold);
+            __syncwarp();
+
+#pragma unroll
+            for (int32_t offset = 1; offset < warp_size; offset = offset << 1)
+            {
+                xdrop_done |= __shfl_up_sync(0xFFFFFFFF, xdrop_done, offset);
+            }
+
+            if (lane_id == warp_size - 1)
+            {
+
+                if (max_pos > prev_max_pos[warp_id])
+                    new_max_found[warp_id] = true;
```

Comment:
Why not simply `new_max_found[warp_id] = (max_pos > prev_max_pos[warp_id])`?

Reply:
Done.